src/theory/combination_care_graph.cpp

```diff
@@ -0,0 +1,93 @@
+/*********************                                                        */
+/*! \file combination_care_graph.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Andrew Reynolds
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2020 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Management of a care graph based approach for theory combination.
+ **/
+
+#include "theory/combination_care_graph.h"
+
+#include "expr/node_visitor.h"
+#include "theory/care_graph.h"
+#include "theory/theory_engine.h"
+
+namespace CVC4 {
+namespace theory {
+
+CombinationCareGraph::CombinationCareGraph(
+    TheoryEngine& te, const std::vector<Theory*>& paraTheories)
+    : CombinationEngine(te, paraTheories)
+{
+}
+
+CombinationCareGraph::~CombinationCareGraph() {}
+
+void CombinationCareGraph::combineTheories()
+{
+  Trace("combineTheories") << "TheoryEngine::combineTheories()" << std::endl;
+
+  // Care graph we'll be building
+  CareGraph careGraph;
+
+  // get the care graph from the parametric theories
+  for (Theory* t : d_paraTheories)
+  {
+    t->getCareGraph(&careGraph);
+  }
+
+  Trace("combineTheories")
+      << "TheoryEngine::combineTheories(): care graph size = "
+      << careGraph.size() << std::endl;
+
+  // Now add splitters for the ones we are interested in
+  CareGraph::const_iterator care_it = careGraph.begin();
+  CareGraph::const_iterator care_it_end = careGraph.end();
+
```

Comment:
```suggestion
```

Reply:
This is necessary in the for loop below.